alcs-api/src/application-reconsideration/application-reconsideration.service.ts

```diff
@@ -0,0 +1,238 @@
+import { CardService } from '../card/card.service';
+import { CodeService } from '../code/code.service';
+import {
+  ServiceNotFoundException,
+  ServiceValidationException,
+} from '../common/exceptions/base.exception';
+import { ApplicationReconsideration } from './application-reconsideration.entity';
+import {
+  ApplicationReconsiderationCreateDto,
+  ApplicationReconsiderationDto,
+  ApplicationReconsiderationUpdateDto,
+  ApplicationReconsiderationWithoutApplicationDto,
+} from './applicationReconsideration.dto';
+
+@Injectable()
+export class ApplicationReconsiderationService {
+  constructor(
+    @InjectRepository(ApplicationReconsideration)
+    private reconsiderationRepository: Repository<ApplicationReconsideration>,
+    @InjectMapper() private mapper: Mapper,
+    private codeService: CodeService,
+    private applicationService: ApplicationService,
+    private cardService: CardService,
+  ) {}
+
+  private DEFAULT_RECONSIDERATION_RELATIONS: FindOptionsRelations<ApplicationReconsideration> =
+    {
+      application: {
+        type: true,
+        region: true,
+        localGovernment: true,
+        decisionMeetings: true,
+      },
+      card: {
+        board: true,
+        type: true,
+        status: true,
+        assignee: true,
+      },
+      type: true,
+    };
+
+  async getByBoardCode(boardCode: string) {
+    return this.reconsiderationRepository.find({
+      where: { card: { board: { code: boardCode } } },
+      relations: this.DEFAULT_RECONSIDERATION_RELATIONS,
+    });
+  }
+
+  async mapToDtos(
+    reconsiderations: ApplicationReconsideration[],
+  ): Promise<ApplicationReconsiderationDto[]> {
+    return this.mapper.mapArrayAsync(
+      reconsiderations,
+      ApplicationReconsideration,
+      ApplicationReconsiderationDto,
+    );
+  }
+
+  async mapToDtosWithoutApplication(
+    reconsiderations: ApplicationReconsideration[],
+  ): Promise<ApplicationReconsiderationWithoutApplicationDto[]> {
+    return this.mapper.mapArrayAsync(
+      reconsiderations,
+      ApplicationReconsideration,
+      ApplicationReconsiderationWithoutApplicationDto,
+    );
+  }
+
+  async create(
+    reconsideration: ApplicationReconsiderationCreateDto,
+    board: Board,
+  ) {
+    const type = await this.fetchAndValidateType(reconsideration.reconTypeCode);
+
+    const newReconsideration = new ApplicationReconsideration({
+      submittedDate: new Date(reconsideration.submittedDate),
+    });
+
+    const newCard = await this.cardService.create(
+      {
+        boardCode: reconsideration.boardCode,
+        typeCode: 'RECON',
+      } as CardCreateDto,
+      board,
+      false,
+    );
+
+    newReconsideration.card = newCard;
+    newReconsideration.type = type;
+
+    newReconsideration.application = await this.getApplication(reconsideration);
+
+    const recon = await this.reconsiderationRepository.save(newReconsideration);
+    return this.getByUuid(recon.uuid);
+  }
+
+  private async getApplication(
```

Comment:
rename getOrCreateApplication just to be clear it can also create

Reply:
yah, I also thought about that but did not like the idea that it is not persisting application on creation, it only prepares the object.  Anyway,  I'll rename it